tests/uploads_test.js

```diff
@@ -3,38 +3,215 @@
+    //     // form,
+    //     {
+    //       file: fs.readFile("tests/assets/lobster-family.csv", (err, data) => {
+    //         if (err) throw err;
+    //         return data;
+    //       }),
+    //     },
+    //     {
+    //       // headers: {...prism.authHeader, "content-type": `multipart/form-data; boundary=${form._boundary}`},
+    //       headers: prism.authHeader
+    //     }
+    //   )
+    // );
+    //
+    // t.assert(response.status === 202);
+
+    // create export
+    response = await prism.setup().then((client) =>
+      client.post(
+        `${resource_endpoint}/${upl_id}/exports`,
+        {
+          type: "all",
+        },
+        {
+          headers: prism.authHeader,
+        }
+      )
+    );
+
+    t.assert(response.status === 200);
+    const ex_id = response.data.exportId;
+
+    // retrieve export
+    response = await prism.setup().then((client) =>
+      client.get(`${resource_endpoint}/${upl_id}/exports/${ex_id}`, {
+        headers: prism.authHeader,
+      })
+    );
+
+    t.assert(response.status === 200);
+
+    // delete
+    response = await prism.setup().then((client) =>
+      client.delete(`${resource_endpoint}/${upl_id}`, {
+        headers: prism.authHeader,
+      })
+    );
+
+    t.assert(response.status === 204);
+  } catch (prismError) {
+    if (Object.keys(prismError).length > 0) {
+      t.fail(JSON.stringify(prismError, null, 2));
+    } else {
+      t.fail(prismError.toString());
+    }
+  }
+});
+
+test("list uploads", async function (t) {
+  try {
+    const response = await prism.setup().then((client) =>
+      client.get(
+        `${resource_endpoint}?${new URLSearchParams({
+          campaignId: t.context.cmp_id,
+        }).toString()}`,
+        {
+          headers: prism.authHeader,
+        }
+      )
+    );
+    if (response.status !== 200) console.log(response);
+    t.assert(response.status === 200);
+    return response.data;
   } catch (prismError) {
     if (Object.keys(prismError).length > 0) {
-      // if the error has properties, it can be JSON.stringified
       t.fail(JSON.stringify(prismError, null, 2));
     } else {
       t.fail(prismError.toString());
     }
   }
 });
 
-// add any tests checking input variations here
+test("errors when columnMapping is not passed in", async function (t) {
+  try {
+    const response = await prism.setup({ errors: false }).then((client) =>
+      client.post(
+        resource_endpoint,
+        {
+          campaignId: t.context.cmp_id,
+        },
+        { headers: prism.authHeader }
+      )
+    );
+
+    t.assert(response.status === 400);
+    t.assert(response.data.message.includes("Invalid body"));
+  } catch (prismError) {
+    if (Object.keys(prismError).length > 0) {
+      t.fail(JSON.stringify(prismError, null, 2));
+    } else {
+      t.fail(prismError.toString());
+    }
+  }
+});
 
-// add any failure cases you need here
+test.after.always("delete campaign", async function (t) {
```

Comment:
**[non-blocking]:** We delete a campaign after the test is done. Should we delete the upload as well?

Reply:
Sorry I didn't see this sooner. The upload is deleted within the same test that creates it. That code can be found [here](https://github.com/lob/lob-openapi/blob/7999515617b79b1c2c5450457724b2571d706758/tests/uploads_test.js#L138-L143). Let me know if I missed something. Thanks for the review :)